src/TestProblem/Hydro/CCSN/dtSolver_CCSN.cpp

```diff
@@ -87,45 +81,34 @@ double Mis_GetTimeStep_Lightbulb( const int lv, const double dTime_dt )
             const real Emag = NULL_REAL;
 #           endif
 
-            const real Dens_Code = Dens;
             const real Eint_Code = Hydro_Con2Eint( Dens, Momx, Momy, Momz, Engy, true, MIN_EINT, Emag );
-            const real Ye        = YeDens / Dens;
 
 
-//          compute the neutrino heating rate
-#           if ( NUC_TABLE_MODE == NUC_TABLE_MODE_TEMP )
-            const int  NTarget = 2;
-#           else
-            const int  NTarget = 3;
-#           endif
-                  int  In_Int[NTarget+1];
-                  real In_Flt[3], Out[NTarget+1];
-
-            In_Flt[0] = Dens_Code;
-            In_Flt[1] = Eint_Code;
-            In_Flt[2] = Ye;
-
-            In_Int[0] = NTarget;
-            In_Int[1] = NUC_VAR_IDX_XN;
-            In_Int[2] = NUC_VAR_IDX_XP;
-#           if ( NUC_TABLE_MODE == NUC_TABLE_MODE_ENGY )
-            In_Int[3] = NUC_VAR_IDX_EORT;
-#           endif
+//          call Src_Lightbulb() to get the heating/cooling rate if not computed yet
+            if ( dEint_Code == NULL_REAL )
+            {
+//             get the input arrays
+               real fluid[FLU_NIN_S];
+
+               for (int v=0; v<FLU_NIN_S; v++)  fluid[v] = amr->patch[ amr->FluSg[lv] ][lv][PID]->fluid[v][k][j][i];
+
+#              ifdef MHD
+               real B[NCOMP_MAG] = { Emag, 0.0, 0.0 };
+#              else
+               real *B = NULL;
+#              endif
 
-            EoS_General_CPUPtr( NUC_MODE_ENGY, Out, In_Flt, In_Int, EoS_AuxArray_Flt, EoS_AuxArray_Int, h_EoS_Table );
 
-            const real Xn       = Out[0];               // neutron mass fraction
-            const real Xp       = Out[1];               // proton  mass fraction
-            const real Temp_MeV = Out[2] * Kelvin2MeV;  // temperature in MeV
+               SrcTerms.Lightbulb_FuncPtr( fluid, B, &SrcTerms, NULL_REAL, dh, x, y, z, NULL_REAL, NULL_REAL,
+                                           MIN_DENS, MIN_PRES, MIN_EINT, &EoS,
+                                           SrcTerms.Lightbulb_AuxArrayDevPtr_Flt, SrcTerms.Lightbulb_AuxArrayDevPtr_Int );
 
-            const double rate_heating = 1.544e20 * ( SrcTerms.Lightbulb_Lnue / 1.0e52 ) * ( 1.0e14 / r2_CGS )
-                                      * SQR( 0.25 * SrcTerms.Lightbulb_Tnue );
-            const double rate_cooling = 1.399e20 * CUBE(  SQR( 0.5 * Temp_MeV )  );
+#              ifdef DELE
+               dEint_Code = fluid[DELE];
+#              endif
+            } // if ( dEint_Code == NULL_REAL )
 
-            const double tau      = 1.0e-11 * Dens_Code * UNIT_D;
-            const double rate_CGS = ( rate_heating - rate_cooling ) * ( Xn + Xp ) * exp( -tau );
 
-            const double dEint_Code  = rate_CGS * UNIT_T * sEint2Code * Dens_Code;  // dEint per UNIT_T
             const double _Eint_Ratio = fabs( dEint_Code / Eint_Code );
```

Comment:
- [ ] Use `FABS` instead of `fabs`.
- [ ] Rename `_Eint_Ratio` as `dt_LB_Inv_ThisCell`